.github/workflows/backport.yml

```diff
@@ -72,21 +72,77 @@ jobs:
 
       - name: Get branch histories
         run: |
-          git fetch origin main --unshallow
+          git fetch origin main --depth 100
+          git fetch origin ${{matrix.target_branch}} --depth 1
           git status
           git log -1 --format='%H'
 
-      - name: Backport commit
+      - name: Checkout the commit into the staging area
         run: |
-          echo "Cherry-pick from $GITHUB_SHA to ${{matrix.target_branch}}"
-          git cherry-pick -x ${{github.event.pull_request.merge_commit_sha}}
+          # Checkout the merged commit
+          git checkout ${{github.event.pull_request.merge_commit_sha}}
 
-          # See https://github.com/elastic/detection-rules/issues/1171
-          # Eventually, this cherry pick command will be:
-          # git-cherry-pick --no-commit
-          # <python code to remove irrelevant rules>
-          # git commit --author ... --message ...
+          # Move it to the staging area
+          git reset --soft HEAD^
 
-      - name: Push changes
+      - name: Install dependencies
         run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt -r requirements-dev.txt
+
+      - name: Prune non-${{matrix.target_branch}} rules
+        env:
+          UNSTAGED_LIST_FILE: "../unstaged-rules.txt"
+        run: |
+          python -m detection_rules dev unstage-incompatible-rules --target-stack-version ${{matrix.target_branch}}
+
+          # Track which rules were unstaged
+          git diff --name-only > $UNSTAGED_LIST_FILE
+
+          # Since they've been tracked, remove any untracked files
+          git checkout -- .
+
+      - name: Commit and push to ${{matrix.target_branch}}
+        env:
+          COMMIT_MSG_FILE: "../commit-message.txt"
+          UNSTAGED_LIST_FILE: "../unstaged-rules.txt"
+        run: |
+          set -x
+
+          echo "Switch to the target branch and keep the staged changes"
+          git checkout ${{matrix.target_branch}}
+
+          NEEDS_BACKPORT=$(git diff HEAD --quiet --exit-code && echo y || echo n)
```

Comment:
nit: 

NEEDS_BACKPORT == y when _not_ needed? `true` or `false` could be more inutitive

Reply:
🤦

Reply:
f7a497d